### ISSUE TITLE
DOI: Better handling of pages with many DOIs

### ISSUE THREAD

**Issue by dstillman** [zotero/translators#3311]
https://forums.zotero.org/discussion/114892/zotero-connector-not-saving

https://journals.ametsoc.org/view/journals/phoc/53/1/JPO-D-22-0001.1.xml

The translator finds 84 DOIs on this page, and then proceeds to download data for all of them in rapid succession, and then one returns a 404 and that seems to fail the whole process.

We should 1) maybe put a more reasonable limit on the number of DOIs we try to get data for, 2) check Crossref's guidance for rate-limiting and see if we're within it (though it may be hard to find guidance on content negotiation specifically rather than the API), 3) check if the connector/translate is failing incorrectly if lookup for a single DOI on the page fails (@adomasven), and 4) merge https://github.com/zotero/translators/pull/3009 if it adds a translator or this page.

**Comment by adomasven** [comment]
In my estimation this `await` https://github.com/zotero/translators/commit/7c4cc221d9348a2d5ae57615053a1c92f104e30f#diff-df0ac0c2a1d09b7146a41c727101b54ea5acf3f7f1dec1775adc9675bb05e361L217 broke the DOI translator such that one retrieval failure breaks translation with DOI completely, because it causes a throw there.

**Comment by dstillman** [comment]
Yep, looks like it. So for starters we probably want to add a try/catch there that just logs the error and continues on, which seems like it was the intended behavior previously (`translate.setHandler("error", function () {});`).

**Comment by dstillman** [comment]
For https://journals.ametsoc.org/view/journals/phoc/53/1/JPO-D-22-0001.1.xml ? I'm not getting EM there.

Works now with PubFactory, though.

**Comment by dstillman** [comment]
Oh wait, maybe this is MV3 breakage. I'm not getting EM in Chrome MV3. Shows up in Firefox and Edge.

**Comment by dstillman** [comment]
Firefox:

```
(4)(+0000309): Translate: Parsing code for Embedded Metadata (951c027d-74ac-47d4-a107-9c3069ab7b48, 2024-03-27 20:15:00)

(3)(+0000000): Translate: Prefix 'og' => 'http://ogp.me/ns#'

(3)(+0000001): Translate: Prefix 'fb' => 'http://ogp.me/ns/fb#'

(3)(+0000000): Translate: Prefix 'article' => 'http://ogp.me/ns/article#'

(3)(+0000000): Translate: Embedded Metadata: found 121 meta tags.
```

Chrome MV3:

```
(4)(+0000006): Translate: Parsing code for Embedded Metadata (951c027d-74ac-47d4-a107-9c3069ab7b48, 2024-03-27 20:15:00)

(3)(+0000000): Translate: Prefix 'og' => 'http://ogp.me/ns#'

(3)(+0000001): Translate: Prefix 'fb' => 'http://ogp.me/ns/fb#'

(3)(+0000000): Translate: Prefix 'article' => 'http://ogp.me/ns/article#'

(3)(+0000000): Translate: Embedded Metadata: found 0 meta tags.
